var/spack/repos/builtin/packages/qscintilla/package.py

```diff
@@ -0,0 +1,76 @@
+# Copyright 2013-2019 Lawrence Livermore National Security, LLC and other
+# Spack Project Developers. See the top-level COPYRIGHT file for details.
+#
+# SPDX-License-Identifier: (Apache-2.0 OR MIT)
+
+from spack import *
+import os
+
+class Qscintilla(QMakePackage):
+    """QScintilla is a port to Qt of Neil Hodgson's Scintilla C++ editor control. """
+
+    homepage = "https://www.riverbankcomputing.com/software/qscintilla/intro"
+    url      = "https://www.riverbankcomputing.com/static/Downloads/QScintilla/2.11.2/QScintilla_gpl-2.11.2.tar.gz"
+
+    version('2.11.2', sha256='029bdc476a069fda2cea3cd937ba19cc7fa614fb90578caef98ed703b658f4a1')
+    # didn't have much luck with newer versions of Qscintilla and QT@4.8.6, so prefer the following version
+    version('2.10.2', sha256='14b31d20717eed95ea9bea4cd16e5e1b72cee7ebac647cba878e0f6db6a65ed0', preferred=True)
+
+    variant('python', default=True, description="Enable python bindings")
+    variant('designer', default=False, description="Enable pluging for Qt-Designer")
+
+    # QScintilla so far tested to compile with Qt@4.8.6
+    depends_on('qt') # qt is not compiling with +phonon +dbus variants enabled
+
+    # Beyond py-pyqt@4.12.1, pyqt4 needs its own sip module (not implemented yet)
+    # Without private sip moduele, python bindings will not compile
+    # Ref: https://www.riverbankcomputing.com/static/Docs/PyQt4/installation.html   
+    # TODO implement private sip module for py-pyqt4
+    depends_on('py-pyqt4@:4.12.1', type='build', when='@qt4') 
+    depends_on('py-pyqt5', type='build', when='qt@5')
+    depends_on('python', type=('build', 'run'))
+
+    # with qt@4.8.6, didn't have much luck in compiling newer versions
+    conflicts('qt@4', when='@2.10.3:') 
+
+    def chdir(self):
+        os.chdir(str(self.stage.source_path)+'/Qt4Qt5')
+
+
+    def qmake_args(self):
+        # below, DEFINES ... gets rid of ...regex...errors during build
+        # although, there shouldn't be such errors since we use '-std=c++11'
+        args = ['CONFIG+=-std=c++11', 'DEFINES+=NO_CXX11_REGEX=1']
+        return args
+
+
+    # the following installs files under "/qscintilla_prefix/qt_prefix"
+    # it gets rid of 'Nothing installed error'
+    # without it libqscintilla is installed under qt_prefix, is that how it should be?
+    def setup_environment(self, spack_env, run_env):
+        spack_env.set('INSTALL_ROOT', prefix)
+
+
+    def make_python_bindings(self):
+        os.chdir(str(self.stage.source_path)+'/Python')
+        python = which('python')
+        if self.version < Version('5'):
+            python('configure.py')
+        else:
+            python('configure.py -pyqt=PyQt5')
```

Comment:
Are you sure this isn't a SIP package, not a QMake package?

Reply:
Not sure. During creation of package via spack, SIPpackage, or qmakepackage classes weren't suggested, I think it is because there are no clues about those in the top directory.

Reply:
There are if you update to the latest version of Spack.

Reply:
I would try switching to SIPPackage

Reply:
I checked that there is a sip folder at:
```
/home/sbulut/Downloads/QScintilla_gpl-2.10.2/Python/
```
but not under 
```
/home/sbulut/Downloads/QScintilla_gpl-2.10.2/Qt4Qt5
```
I think that means Python bindings are of 'sippackage' type while libqscintilla is of 'qmakepackage' type. Is it possible to switch to sippackage class halfway thorugh?

Reply:
Hmm, nope. Just see how the documentation recommends building the package. If they use `configure.py`, `make`, `make install`, then use SIPPackage. If they use `qmake`, `make`, `make install`, then use QMakePackage.

Reply:
They suggest using qmake .. for QScintilla and 'Python configure ..' for Python bindings as indicated in this ref: https://www.riverbankcomputing.com/static/Docs/QScintilla/
May be we could create an artificial SIPpackage for Python bindings only, and make it a dependency or a default variant for qscindilla.